x-pack/legacy/plugins/reporting/server/browsers/network_policy.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as _ from 'lodash';
+import * as url from 'url';
+
+interface FirewallRequest {
+  ip: string | null;
+  url: string;
+}
+
+interface FirewallRule {
+  allow: boolean;
+  hosts?: string[];
+  protocols?: string[];
+  ips?: string[];
+}
+
```

Comment:
This is my network-policy opus, you can do anything in terms of allowing/denying requests, including OR-ing a particular filter (see tests).